### PR TITLE
Feature/python3.8

### LINKED ISSUE #59: [Feature] Multithreading feature in python3 for corresponding python2 file
The `master` branch is maintained for *python2* whereas the `feature/python3.8` branch is maintained for *python3*. Multithreading feature is available in *python2* as of now. 

The multithreading feature has been used in the project to improve efficiency of the scanning process. Checkout 'Need for Multithreading' section in  [README.md](https://github.com/vinitshahdeo/PortScanner#need-for-multithreading-white_check_mark) for better understanding.

Create a PR in `feature/python3.8` branch for the multithreading feature in *python3*.

The file correspondence in *python2* - *python3*(`master` - `feature/python3.8`) can be understood as follows:
1. `src/single/scanner.py` - `scanner.py` ( Port scanning feature without multithreading )
2. `src/multi/scanner_thread.py` - _(to be written in py3)_  *Split processing* function for multithreading.
3. `src/mainScanner.py` - _(to be written in py3)_ Port Scanning feature using multithreading.
4. `src/ipscanner` - _(to be written in py3)_ IP Scanning feature using multithreading.

Follow the same directory structure as in `master`.

### PR DESCRIPTION
**Pull request by tanay13**
# Description
Followed the same directory structure as in master branch .
added split processing function in feature/python3.8 branch
Multithreading enabled in python 3 .
Fixes #59  (issue)

## Type of change
converted python 2 file to python 3 

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have added all the three reviewers: @vinitshahdeo, @kashish121 & @ishika1727

### CODE REVIEW

**Comment by tanay13** [comment]
Please review this @vinitshahdeo , @Kashish121 and @ishika1727

File: src/ipscanner.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+
+# Check what time the scan started
+td1 = datetime.now()
+
+try:
+    with open(get_absolute_path('../config.json')) as config_file:
+        config = json.load(config_file)
+        # print get_absolute_path('../config.json')
+    range_low = int(config['ipRange']['low'])
```
Comment (Kashish121) [inline src/ipscanner.py:36]: For `ipScanner.py` to read values *ipRange.low* and *ipRange.high* from config.json, these values should actually be present inside the config.json file.
You will have to update the *config.json* in `feature/python3.8`. Consider copying the *config.json* file from `master` branch.

File: src/ipscanner.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    range_high = int(config['ipRange']['high'])
```
Comment (Kashish121) [inline src/ipscanner.py:37]: Same with this, update *config.json*.

File: src/mainScanner.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+from multi.scanner_thread import split_processing
+import logging
+from flask import Flask, render_template, request, redirect, url_for
+
+app = Flask(__name__)
+
+
+@app.route('/')
+def homepage():
+    return render_template('index.html')
```
Comment (Kashish121) [inline src/mainScanner.py:19]: For this to actually work and render the page, you need to add `index.html` inside your directory.
Consider copying the `src/static` and `src/templates` folder contents in this branch.

File: src/mainScanner.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+
+exc = getattr(__builtin__, "IOError", "FileNotFoundError")
```
Comment (Kashish121) [inline src/mainScanner.py:22]: __builtin__ won't work for *python3*. Use builtins instead.

**Comment by tanay13** [comment]
@Kashish121  I corrected everything please have a look

**Review by vinitshahdeo — COMMENTED** [review]
@tanay13 I can see frontend changes in this PR?

> @Kashish121 As per the scope of this PR, there shouldn't be any changes in `src/static`? Please check - Probably @tanay13 has branched from the wrong branch. Please guide him further.

**Comment by tanay13** [comment]
@Kashish121 please tell me how to proceed

**Pull request merged**